### PR TITLE
3.x How to disable auto-tables

### PR DESCRIPTION
**Pull request by inoas**
Ref: https://github.com/cakephp/cakephp/issues/10302

### CODE REVIEW

**Comment by inoas** [comment]
Anything to change?

File: en/development/configuration.rst [hunk 1]
```diff
@@ -681,6 +681,28 @@ use the values that are defined in the **app.php**. You can use
 environment variables in a local development. See the Readme instructions of the
 library for more information.
 
+
+Disabling Auto-Tables
+=====================
+
+While auto-tables are really useful when quickly creating new applications and baking models, they can be hard to debug.
```
Comment (ravage84) [inline en/development/configuration.rst]: hard -> difficult

Comment (ravage84) [inline en/development/configuration.rst]: Please shorten the line lengths.

File: en/development/configuration.rst [hunk 2]
```diff
@@ -681,6 +681,28 @@ use the values that are defined in the **app.php**. You can use
...
+
+You can see if any query was fired of auto-tables when you go to history tab, select a request and then go to the sql tab in DebugKit.
```
Comment (ravage84) [inline en/development/configuration.rst]: "when you go to" -> "on the"

sql -> SQL

Comment (ravage84) [inline en/development/configuration.rst]: Can you reword "of auto-tables"?

File: en/development/configuration.rst [hunk 3]
```diff
@@ -681,6 +681,28 @@ use the values that are defined in the **app.php**. You can use
...
 
+
+Disabling Auto-Tables
+=====================
+
+While auto-tables are really useful when quickly creating new applications and baking models, they can be hard to debug.
+
+You can see if any query was fired of auto-tables when you go to history tab, select a request and then go to the sql tab in DebugKit. 
+
+If that is not sufficient, you can throw an exeception if CakePHP is working on ``Cake\ORM\Table``, aka with auto-tables instead of ``App\ORM\YourConcreteExampleTable`` class like so::
```
Comment (ravage84) [inline en/development/configuration.rst]: The explanation what "auto-tables" are should probably be first.

Comment (ravage84) [inline en/development/configuration.rst]: aka -> a.k.a., makes it more clear for non-English native speakers.

File: en/development/configuration.rst [hunk 4]
```diff
@@ -681,6 +681,28 @@ use the values that are defined in the **app.php**. You can use
...
+
+    // In your bootstrap.php
+    use Cake\Event\EventManager;
+    use Cake\Network\Exception\InternalErrorException;
+    
+    // Disable Auto-Tables
+    EventManager::instance()->on('Model.initialize', function($event) {
+        if (get_class($event->getSubject()) === 'Cake\ORM\Table') {
```
Comment (ravage84) [inline en/development/configuration.rst]:
`````php
$subject = ($event->getSubject();
`````

is used twice.

File: en/development/configuration.rst (hunk 2 above)
Comment (inoas) [inline en/development/configuration.rst]: automatic tables? dummy table classes? placeholder table classes? generic table classes?
any ideas @lorenzo ?

File: en/development/configuration.rst (hunk 3 above)
Comment (inoas) [inline en/development/configuration.rst]: e.g.?

File: en/development/configuration.rst (hunk 2 above)
Comment (inoas) [inline en/development/configuration.rst]: I hope generic table class is good.

**Comment by inoas** [comment]
- aka -> e.g.
- attribute off not genitive of
- $subject var
- wording
- formatting
- meta keywords

Good to merge?

**Comment by inoas** [comment]
Note: https://github.com/AD7six/cakephp-shadow-translate fires off generic tables by default and does not work if they are disabled until you bake the minimum tables.

**Pull request merged**